internal/controller/finbackupconfig_controller_test.go

```diff
@@ -0,0 +1,330 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	finv1 "github.com/cybozu-go/fin/api/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func newControllerPod(name, namespace string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: "test-sa",
+			Containers: []corev1.Container{
+				{
+					Name:  "manager",
+					Image: "test:latest",
+				},
+			},
+		},
+	}
+}
+
+func findEnvVar(envVars []corev1.EnvVar, name string) *corev1.EnvVar {
+	for i := range envVars {
+		if envVars[i].Name == name {
+			return &envVars[i]
+		}
+	}
+	return nil
+}
+
+var _ = Describe("FinBackupConfig Controller", func() {
+	var reconciler *FinBackupConfigReconciler
+	var ctx context.Context
+	var fbcNamespace string
+
+	BeforeEach(func() {
+		ctx = context.Background()
+
+		fbcNamespace = fmt.Sprintf("%s-fbc-%d", namespace, time.Now().UnixNano())
+
+		reconciler = NewFinBackupConfigReconciler(
+			k8sClient,
+			scheme.Scheme,
+			"",
+			namespace,
+		)
```

Comment:
The third test case doesn't use this, so let's move this to each test case body.